Build failure: ardour

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

Build *ardour*

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
...
[ 726/1969] Compiling libs/ardour/minibpm.cc
[ 727/1969] Compiling libs/ardour/mididm.cc
[ 728/1969] Compiling libs/ardour/midiport_manager.cc
[ 729/1969] Compiling libs/ardour/source.cc
[ 730/1969] Compiling libs/ardour/speakers.cc
[ 731/1969] Compiling libs/ardour/midi_ui.cc
[ 732/1969] Compiling libs/ardour/midi_track.cc
[ 733/1969] Compiling libs/ardour/source_factory.cc
[ 734/1969] Compiling libs/ardour/midi_state_tracker.cc
[ 735/1969] Compiling libs/ardour/midi_stretch.cc
[ 736/1969] Compiling libs/ardour/midi_source.cc
[ 737/1969] Compiling libs/ardour/midi_scene_changer.cc
[ 738/1969] Compiling libs/ardour/soundcloud_upload.cc
[ 739/1969] Compiling libs/ardour/midi_scene_change.cc
[ 740/1969] Compiling libs/ardour/midi_ring_buffer.cc
[ 741/1969] Compiling libs/ardour/transpose.cc
[ 742/1969] Compiling libs/ardour/midi_region.cc
[ 743/1969] Compiling libs/ardour/midi_port.cc
[ 744/1969] Compiling libs/ardour/solo_isolate_control.cc
[ 745/1969] Compiling libs/ardour/solo_mute_release.cc
[ 746/1969] Compiling libs/ardour/midi_playlist.cc
[ 747/1969] Compiling libs/ardour/midi_patch_manager.cc
[ 748/1969] Compiling libs/ardour/midi_model.cc
[ 749/1969] Compiling libs/ardour/solo_control.cc
[ 750/1969] Compiling libs/ardour/solo_safe_control.cc
[ 751/1969] Compiling libs/ardour/midi_clock_slave.cc
[ 752/1969] Compiling libs/ardour/midi_automation_list_binder.cc
../libs/ardour/soundcloud_upload.cc: In member function ‘std::string SoundcloudUploader::Get_Auth_Token(std::string, std::string)’:
../libs/ardour/soundcloud_upload.cc:77:25: warning: ‘CURLFORM_COPYNAME’ is deprecated: since 7.56.0. Use curl_mime_name() [-Wdeprecated-declarations]
   77 |                         CURLFORM_COPYNAME, "client_id",
      |                         ^~~~~~~~~~~~~~~~~
In file included from ../libs/ardour/ardour/soundcloud_upload.h:17,
                 from ../libs/ardour/soundcloud_upload.cc:21:
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2524:3: note: declared here
 2524 |   CURLFORM_COPYNAME        CURL_DEPRECATED(7.56.0, "Use curl_mime_name()"),
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:78:25: warning: ‘CURLFORM_COPYCONTENTS’ is deprecated: since 7.56.0. Use curl_mime_data() [-Wdeprecated-declarations]
   78 |                         CURLFORM_COPYCONTENTS, "6dd9cf0ad281aa57e07745082cec580b",
      |                         ^~~~~~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2527:3: note: declared here
 2527 |   CURLFORM_COPYCONTENTS    CURL_DEPRECATED(7.56.0, "Use curl_mime_data()"),
      |   ^~~~~~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:75:21: warning: ‘CURLFORMcode curl_formadd(curl_httppost**, curl_httppost**, ...)’ is deprecated: since 7.56.0. Use curl_mime_init() [-Wdeprecated-declarations]
   75 |         curl_formadd(&formpost,
      |         ~~~~~~~~~~~~^~~~~~~~~~~
   76 |                         &lastptr,
      |                         ~~~~~~~~~
   77 |                         CURLFORM_COPYNAME, "client_id",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   78 |                         CURLFORM_COPYCONTENTS, "6dd9cf0ad281aa57e07745082cec580b",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   79 |                         CURLFORM_END);
      |                         ~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2599:1: note: declared here
 2599 | curl_formadd(struct curl_httppost **httppost,
      | ^~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:83:25: warning: ‘CURLFORM_COPYNAME’ is deprecated: since 7.56.0. Use curl_mime_name() [-Wdeprecated-declarations]
   83 |                         CURLFORM_COPYNAME, "client_secret",
      |                         ^~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2524:3: note: declared here
 2524 |   CURLFORM_COPYNAME        CURL_DEPRECATED(7.56.0, "Use curl_mime_name()"),
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:84:25: warning: ‘CURLFORM_COPYCONTENTS’ is deprecated: since 7.56.0. Use curl_mime_data() [-Wdeprecated-declarations]
   84 |                         CURLFORM_COPYCONTENTS, "53f5b0113fb338800f8a7a9904fc3569",
      |                         ^~~~~~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2527:3: note: declared here
 2527 |   CURLFORM_COPYCONTENTS    CURL_DEPRECATED(7.56.0, "Use curl_mime_data()"),
      |   ^~~~~~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:81:21: warning: ‘CURLFORMcode curl_formadd(curl_httppost**, curl_httppost**, ...)’ is deprecated: since 7.56.0. Use curl_mime_init() [-Wdeprecated-declarations]
   81 |         curl_formadd(&formpost,
      |         ~~~~~~~~~~~~^~~~~~~~~~~
   82 |                         &lastptr,
      |                         ~~~~~~~~~
   83 |                         CURLFORM_COPYNAME, "client_secret",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   84 |                         CURLFORM_COPYCONTENTS, "53f5b0113fb338800f8a7a9904fc3569",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   85 |                         CURLFORM_END);
      |                         ~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2599:1: note: declared here
 2599 | curl_formadd(struct curl_httppost **httppost,
      | ^~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:89:25: warning: ‘CURLFORM_COPYNAME’ is deprecated: since 7.56.0. Use curl_mime_name() [-Wdeprecated-declarations]
   89 |                         CURLFORM_COPYNAME, "grant_type",
      |                         ^~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2524:3: note: declared here
 2524 |   CURLFORM_COPYNAME        CURL_DEPRECATED(7.56.0, "Use curl_mime_name()"),
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:90:25: warning: ‘CURLFORM_COPYCONTENTS’ is deprecated: since 7.56.0. Use curl_mime_data() [-Wdeprecated-declarations]
   90 |                         CURLFORM_COPYCONTENTS, "password",
      |                         ^~~~~~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2527:3: note: declared here
 2527 |   CURLFORM_COPYCONTENTS    CURL_DEPRECATED(7.56.0, "Use curl_mime_data()"),
      |   ^~~~~~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:87:21: warning: ‘CURLFORMcode curl_formadd(curl_httppost**, curl_httppost**, ...)’ is deprecated: since 7.56.0. Use curl_mime_init() [-Wdeprecated-declarations]
   87 |         curl_formadd(&formpost,
      |         ~~~~~~~~~~~~^~~~~~~~~~~
   88 |                         &lastptr,
      |                         ~~~~~~~~~
   89 |                         CURLFORM_COPYNAME, "grant_type",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   90 |                         CURLFORM_COPYCONTENTS, "password",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   91 |                         CURLFORM_END);
      |                         ~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2599:1: note: declared here
 2599 | curl_formadd(struct curl_httppost **httppost,
      | ^~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:95:25: warning: ‘CURLFORM_COPYNAME’ is deprecated: since 7.56.0. Use curl_mime_name() [-Wdeprecated-declarations]
   95 |                         CURLFORM_COPYNAME, "username",
      |                         ^~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2524:3: note: declared here
 2524 |   CURLFORM_COPYNAME        CURL_DEPRECATED(7.56.0, "Use curl_mime_name()"),
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:96:25: warning: ‘CURLFORM_COPYCONTENTS’ is deprecated: since 7.56.0. Use curl_mime_data() [-Wdeprecated-declarations]
   96 |                         CURLFORM_COPYCONTENTS, username.c_str(),
      |                         ^~~~~~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2527:3: note: declared here
 2527 |   CURLFORM_COPYCONTENTS    CURL_DEPRECATED(7.56.0, "Use curl_mime_data()"),
      |   ^~~~~~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:93:21: warning: ‘CURLFORMcode curl_formadd(curl_httppost**, curl_httppost**, ...)’ is deprecated: since 7.56.0. Use curl_mime_init() [-Wdeprecated-declarations]
   93 |         curl_formadd(&formpost,
      |         ~~~~~~~~~~~~^~~~~~~~~~~
   94 |                         &lastptr,
      |                         ~~~~~~~~~
   95 |                         CURLFORM_COPYNAME, "username",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   96 |                         CURLFORM_COPYCONTENTS, username.c_str(),
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   97 |                         CURLFORM_END);
      |                         ~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2599:1: note: declared here
 2599 | curl_formadd(struct curl_httppost **httppost,
      | ^~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:101:25: warning: ‘CURLFORM_COPYNAME’ is deprecated: since 7.56.0. Use curl_mime_name() [-Wdeprecated-declarations]
  101 |                         CURLFORM_COPYNAME, "password",
      |                         ^~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2524:3: note: declared here
 2524 |   CURLFORM_COPYNAME        CURL_DEPRECATED(7.56.0, "Use curl_mime_name()"),
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:102:25: warning: ‘CURLFORM_COPYCONTENTS’ is deprecated: since 7.56.0. Use curl_mime_data() [-Wdeprecated-declarations]
  102 |                         CURLFORM_COPYCONTENTS, password.c_str(),
      |                         ^~~~~~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2527:3: note: declared here
 2527 |   CURLFORM_COPYCONTENTS    CURL_DEPRECATED(7.56.0, "Use curl_mime_data()"),
      |   ^~~~~~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:99:21: warning: ‘CURLFORMcode curl_formadd(curl_httppost**, curl_httppost**, ...)’ is deprecated: since 7.56.0. Use curl_mime_init() [-Wdeprecated-declarations]
   99 |         curl_formadd(&formpost,
      |         ~~~~~~~~~~~~^~~~~~~~~~~
  100 |                         &lastptr,
      |                         ~~~~~~~~~
  101 |                         CURLFORM_COPYNAME, "password",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  102 |                         CURLFORM_COPYCONTENTS, password.c_str(),
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  103 |                         CURLFORM_END);
      |                         ~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2599:1: note: declared here
 2599 | curl_formadd(struct curl_httppost **httppost,
      | ^~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:113:39: warning: ‘CURLOPT_HTTPPOST’ is deprecated: since 7.56.0. Use CURLOPT_MIMEPOST [-Wdeprecated-declarations]
  113 |         curl_easy_setopt(curl_handle, CURLOPT_HTTPPOST, formpost);
      |                                       ^~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:1210:3: note: declared here
 1210 |   CURLOPTDEPRECATED(CURLOPT_HTTPPOST, CURLOPTTYPE_OBJECTPOINT, 24,
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc: In member function ‘std::string SoundcloudUploader::Upload(std::string, std::string, std::string, bool, bool, ARDOUR::ExportHandler*)’:
../libs/ardour/soundcloud_upload.cc:175:25: warning: ‘CURLFORM_COPYNAME’ is deprecated: since 7.56.0. Use curl_mime_name() [-Wdeprecated-declarations]
  175 |                         CURLFORM_COPYNAME, "track[asset_data]",
      |                         ^~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2524:3: note: declared here
 2524 |   CURLFORM_COPYNAME        CURL_DEPRECATED(7.56.0, "Use curl_mime_name()"),
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:176:25: warning: ‘CURLFORM_FILE’ is deprecated: since 7.56.0. Use curl_mime_filedata() [-Wdeprecated-declarations]
  176 |                         CURLFORM_FILE, file_path.c_str(),
      |                         ^~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2533:3: note: declared here
 2533 |   CURLFORM_FILE            CURL_DEPRECATED(7.56.0, "Use curl_mime_filedata()"),
      |   ^~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:173:21: warning: ‘CURLFORMcode curl_formadd(curl_httppost**, curl_httppost**, ...)’ is deprecated: since 7.56.0. Use curl_mime_init() [-Wdeprecated-declarations]
  173 |         curl_formadd(&formpost,
      |         ~~~~~~~~~~~~^~~~~~~~~~~
  174 |                         &lastptr,
      |                         ~~~~~~~~~
  175 |                         CURLFORM_COPYNAME, "track[asset_data]",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  176 |                         CURLFORM_FILE, file_path.c_str(),
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  177 |                         CURLFORM_END);
      |                         ~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2599:1: note: declared here
 2599 | curl_formadd(struct curl_httppost **httppost,
      | ^~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:182:25: warning: ‘CURLFORM_COPYNAME’ is deprecated: since 7.56.0. Use curl_mime_name() [-Wdeprecated-declarations]
  182 |                         CURLFORM_COPYNAME, "oauth_token",
      |                         ^~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2524:3: note: declared here
 2524 |   CURLFORM_COPYNAME        CURL_DEPRECATED(7.56.0, "Use curl_mime_name()"),
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:183:25: warning: ‘CURLFORM_COPYCONTENTS’ is deprecated: since 7.56.0. Use curl_mime_data() [-Wdeprecated-declarations]
  183 |                         CURLFORM_COPYCONTENTS, token.c_str(),
      |                         ^~~~~~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2527:3: note: declared here
 2527 |   CURLFORM_COPYCONTENTS    CURL_DEPRECATED(7.56.0, "Use curl_mime_data()"),
      |   ^~~~~~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:180:21: warning: ‘CURLFORMcode curl_formadd(curl_httppost**, curl_httppost**, ...)’ is deprecated: since 7.56.0. Use curl_mime_init() [-Wdeprecated-declarations]
  180 |         curl_formadd(&formpost,
      |         ~~~~~~~~~~~~^~~~~~~~~~~
  181 |                         &lastptr,
      |                         ~~~~~~~~~
  182 |                         CURLFORM_COPYNAME, "oauth_token",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  183 |                         CURLFORM_COPYCONTENTS, token.c_str(),
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  184 |                         CURLFORM_END);
      |                         ~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2599:1: note: declared here
 2599 | curl_formadd(struct curl_httppost **httppost,
      | ^~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:188:25: warning: ‘CURLFORM_COPYNAME’ is deprecated: since 7.56.0. Use curl_mime_name() [-Wdeprecated-declarations]
  188 |                         CURLFORM_COPYNAME, "track[title]",
      |                         ^~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2524:3: note: declared here
 2524 |   CURLFORM_COPYNAME        CURL_DEPRECATED(7.56.0, "Use curl_mime_name()"),
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:189:25: warning: ‘CURLFORM_COPYCONTENTS’ is deprecated: since 7.56.0. Use curl_mime_data() [-Wdeprecated-declarations]
  189 |                         CURLFORM_COPYCONTENTS, title.c_str(),
      |                         ^~~~~~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2527:3: note: declared here
 2527 |   CURLFORM_COPYCONTENTS    CURL_DEPRECATED(7.56.0, "Use curl_mime_data()"),
      |   ^~~~~~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:186:21: warning: ‘CURLFORMcode curl_formadd(curl_httppost**, curl_httppost**, ...)’ is deprecated: since 7.56.0. Use curl_mime_init() [-Wdeprecated-declarations]
  186 |         curl_formadd(&formpost,
      |         ~~~~~~~~~~~~^~~~~~~~~~~
  187 |                         &lastptr,
      |                         ~~~~~~~~~
  188 |                         CURLFORM_COPYNAME, "track[title]",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  189 |                         CURLFORM_COPYCONTENTS, title.c_str(),
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  190 |                         CURLFORM_END);
      |                         ~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2599:1: note: declared here
 2599 | curl_formadd(struct curl_httppost **httppost,
      | ^~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:194:25: warning: ‘CURLFORM_COPYNAME’ is deprecated: since 7.56.0. Use curl_mime_name() [-Wdeprecated-declarations]
  194 |                         CURLFORM_COPYNAME, "track[sharing]",
      |                         ^~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2524:3: note: declared here
 2524 |   CURLFORM_COPYNAME        CURL_DEPRECATED(7.56.0, "Use curl_mime_name()"),
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:195:25: warning: ‘CURLFORM_COPYCONTENTS’ is deprecated: since 7.56.0. Use curl_mime_data() [-Wdeprecated-declarations]
  195 |                         CURLFORM_COPYCONTENTS, ispublic ? "public" : "private",
      |                         ^~~~~~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2527:3: note: declared here
 2527 |   CURLFORM_COPYCONTENTS    CURL_DEPRECATED(7.56.0, "Use curl_mime_data()"),
      |   ^~~~~~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:192:21: warning: ‘CURLFORMcode curl_formadd(curl_httppost**, curl_httppost**, ...)’ is deprecated: since 7.56.0. Use curl_mime_init() [-Wdeprecated-declarations]
  192 |         curl_formadd(&formpost,
      |         ~~~~~~~~~~~~^~~~~~~~~~~
  193 |                         &lastptr,
      |                         ~~~~~~~~~
  194 |                         CURLFORM_COPYNAME, "track[sharing]",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  195 |                         CURLFORM_COPYCONTENTS, ispublic ? "public" : "private",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  196 |                         CURLFORM_END);
      |                         ~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2599:1: note: declared here
 2599 | curl_formadd(struct curl_httppost **httppost,
      | ^~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:200:25: warning: ‘CURLFORM_COPYNAME’ is deprecated: since 7.56.0. Use curl_mime_name() [-Wdeprecated-declarations]
  200 |                         CURLFORM_COPYNAME, "track[downloadable]",
      |                         ^~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2524:3: note: declared here
 2524 |   CURLFORM_COPYNAME        CURL_DEPRECATED(7.56.0, "Use curl_mime_name()"),
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:201:25: warning: ‘CURLFORM_COPYCONTENTS’ is deprecated: since 7.56.0. Use curl_mime_data() [-Wdeprecated-declarations]
  201 |                         CURLFORM_COPYCONTENTS, downloadable ? "true" : "false",
      |                         ^~~~~~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2527:3: note: declared here
 2527 |   CURLFORM_COPYCONTENTS    CURL_DEPRECATED(7.56.0, "Use curl_mime_data()"),
      |   ^~~~~~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:198:21: warning: ‘CURLFORMcode curl_formadd(curl_httppost**, curl_httppost**, ...)’ is deprecated: since 7.56.0. Use curl_mime_init() [-Wdeprecated-declarations]
  198 |         curl_formadd(&formpost,
      |         ~~~~~~~~~~~~^~~~~~~~~~~
  199 |                         &lastptr,
      |                         ~~~~~~~~~
  200 |                         CURLFORM_COPYNAME, "track[downloadable]",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  201 |                         CURLFORM_COPYCONTENTS, downloadable ? "true" : "false",
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  202 |                         CURLFORM_END);
      |                         ~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2599:1: note: declared here
 2599 | curl_formadd(struct curl_httppost **httppost,
      | ^~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:221:47: warning: ‘CURLOPT_HTTPPOST’ is deprecated: since 7.56.0. Use CURLOPT_MIMEPOST [-Wdeprecated-declarations]
  221 |                 curl_easy_setopt(curl_handle, CURLOPT_HTTPPOST, formpost);
      |                                               ^~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:1210:3: note: declared here
 1210 |   CURLOPTDEPRECATED(CURLOPT_HTTPPOST, CURLOPTTYPE_OBJECTPOINT, 24,
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:227:48: warning: ‘CURLOPT_PROGRESSFUNCTION’ is deprecated: since 7.32.0. Use CURLOPT_XFERINFOFUNCTION [-Wdeprecated-declarations]
  227 |                 curl_easy_setopt (curl_handle, CURLOPT_PROGRESSFUNCTION, &SoundcloudUploader::progress_callback);
      |                                                ^~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:1309:3: note: declared here
 1309 |   CURLOPTDEPRECATED(CURLOPT_PROGRESSFUNCTION, CURLOPTTYPE_FUNCTIONPOINT, 56,
      |   ^~~~~~~~~~~~~~~~~
../libs/ardour/soundcloud_upload.cc:287:30: warning: ‘void curl_formfree(curl_httppost*)’ is deprecated: since 7.56.0. Use curl_mime_free() [-Wdeprecated-declarations]
  287 |                 curl_formfree(formpost);
      |                 ~~~~~~~~~~~~~^~~~~~~~~~
/nix/store/azfhp07yysbxvdhm78n43cjiwh1lb111-curl-8.11.1-dev/include/curl/curl.h:2635:1: note: declared here
 2635 | curl_formfree(struct curl_httppost *form);
      | ^~~~~~~~~~~~~

[ 753/1969] Compiling libs/ardour/midi_buffer.cc
[ 754/1969] Compiling libs/ardour/sndfilesource.cc
[ 755/1969] Compiling libs/ardour/meter.cc
[ 756/1969] Compiling libs/ardour/lufs_meter.cc
[ 757/1969] Compiling libs/ardour/luaproc.cc
[ 758/1969] Compiling libs/ardour/luascripting.cc
[ 759/1969] Compiling libs/ardour/sndfile_helpers.cc
[ 760/1969] Compiling libs/ardour/sndfileimportable.cc
[ 761/1969] Compiling libs/ardour/lua_api.cc
[ 762/1969] Compiling libs/ardour/ltc_file_reader.cc
[ 763/1969] Compiling libs/ardour/ltc_slave.cc
[ 764/1969] Compiling libs/ardour/slavable_automation_control.cc
[ 765/1969] Compiling libs/ardour/location_importer.cc
[ 766/1969] Compiling libs/ardour/location.cc
[ 767/1969] Compiling libs/ardour/smf_source.cc
[ 768/1969] Compiling libs/ardour/library.cc
[ 769/1969] Compiling libs/ardour/legatize.cc
[ 770/1969] Compiling libs/ardour/io_tasklist.cc
[ 771/1969] Compiling libs/ardour/ladspa_plugin.cc
[ 772/1969] Compiling libs/ardour/kmeterdsp.cc
[ 773/1969] Compiling libs/ardour/latent.cc
[ 774/1969] Compiling libs/ardour/session_transport.cc
[ 775/1969] Compiling libs/ardour/slavable.cc
[ 776/1969] Compiling libs/ardour/io_processor.cc
[ 777/1969] Compiling libs/ardour/io_plug.cc
[ 778/1969] Compiling libs/ardour/simple_export.cc
[ 779/1969] Compiling libs/ardour/io.cc
[ 780/1969] Compiling libs/ardour/sidechain.cc
[ 781/1969] Compiling libs/ardour/internal_return.cc
[ 782/1969] Compiling libs/ardour/midi_channel_filter.cc
[ 783/1969] Compiling libs/ardour/import_pt.cc
[ 784/1969] Compiling libs/ardour/instrument_info.cc
[ 785/1969] Compiling libs/ardour/import.cc
[ 786/1969] Compiling libs/ardour/iec2ppmdsp.cc
[ 787/1969] Compiling libs/ardour/session_time.cc
[ 788/1969] Compiling libs/ardour/iec1ppmdsp.cc
[ 789/1969] Compiling libs/ardour/graph_edges.cc
[ 790/1969] Compiling libs/ardour/internal_send.cc
[ 791/1969] Compiling libs/ardour/session_state_utils.cc
[ 792/1969] Compiling libs/ardour/globals.cc
[ 793/1969] Compiling libs/ardour/graph.cc
[ 794/1969] Compiling libs/ardour/session_rtevents.cc
[ 795/1969] Compiling libs/ardour/session_state.cc
[ 796/1969] Compiling libs/ardour/gain_control.cc
[ 797/1969] Compiling libs/ardour/fluid_synth.cc
[ 798/1969] Compiling libs/ardour/find_session.cc
[ 799/1969] Compiling libs/ardour/fixed_delay.cc
[ 800/1969] Compiling libs/ardour/session_process.cc
[ 801/1969] Compiling libs/ardour/filter.cc
[ 802/1969] Compiling libs/ardour/filesystem_paths.cc
[ 803/1969] Compiling libs/ardour/file_source.cc
[ 804/1969] Compiling libs/ardour/filename_extensions.cc
[ 805/1969] Compiling libs/ardour/session_midi.cc
[ 806/1969] Compiling libs/ardour/session_playlists.cc
[ 807/1969] Compiling libs/ardour/ffmpegfilesource.cc
[ 808/1969] Compiling libs/ardour/ffmpegfileimportable.cc
[ 809/1969] Compiling libs/ardour/export_timespan.cc
[ 810/1969] Compiling libs/ardour/session_object.cc
[ 811/1969] Compiling libs/ardour/export_smf_writer.cc
[ 812/1969] Compiling libs/ardour/export_status.cc
[ 813/1969] Compiling libs/ardour/export_preset.cc
[ 814/1969] Compiling libs/ardour/export_profile_manager.cc
[ 815/1969] Compiling libs/ardour/graphnode.cc
../libs/ardour/graph.cc: In member function ‘void ARDOUR::Graph::process_one_route(ARDOUR::Route*)’:
../libs/ardour/graph.cc:554:9: warning: ‘retval’ may be used uninitialized [-Wmaybe-uninitialized]
  554 |         if (retval) {
      |         ^~
../libs/ardour/graph.cc:535:14: note: ‘retval’ was declared here
  535 |         int  retval;
      |              ^~~~~~

[ 816/1969] Compiling libs/ardour/element_import_handler.cc
[ 817/1969] Compiling libs/ardour/rt_task.cc
[ 818/1969] Compiling libs/ardour/rb_effect.cc
[ 819/1969] Compiling libs/ardour/session_metadata.cc
[ 820/1969] Compiling libs/ardour/segment_descriptor.cc
[ 821/1969] Compiling libs/ardour/route_group.cc
[ 822/1969] Compiling libs/ardour/delivery.cc
[ 823/1969] Compiling libs/ardour/automatable.cc
[ 824/1969] Compiling libs/ardour/selection.cc
../libs/ardour/globals.cc: In function ‘bool ARDOUR::set_translations_enabled(bool)’:
../libs/ardour/globals.cc:1018:23: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
 1018 |         (void)::write (fd, &c, 1);
      |               ~~~~~~~~^~~~~~~~~~~

[ 825/1969] Compiling libs/ardour/export_graph_builder.cc
[ 826/1969] Compiling libs/ardour/session_export.cc
[ 827/1969] Compiling libs/ardour/export_format_manager.cc
[ 828/1969] Compiling libs/ardour/ebur128_analysis.cc
[ 829/1969] Compiling libs/ardour/disk_reader.cc
[ 830/1969] Compiling libs/ardour/session_events.cc
[ 831/1969] Compiling libs/ardour/export_format_base.cc
[ 832/1969] Compiling libs/ardour/export_channel_configuration.cc
[ 833/1969] Compiling libs/ardour/export_filename.cc
[ 834/1969] Compiling libs/ardour/export_failed.cc
[ 835/1969] Compiling libs/ardour/session_command.cc
[ 836/1969] Compiling libs/ardour/session_bundles.cc
[ 837/1969] Compiling libs/ardour/disk_writer.cc
[ 838/1969] Compiling libs/ardour/dsp_filter.cc
[ 839/1969] Compiling libs/ardour/session.cc
[ 840/1969] Compiling libs/ardour/session_handle.cc
[ 841/1969] Compiling libs/ardour/session_configuration.cc
[ 842/1969] Compiling libs/ardour/export_handler.cc
[ 843/1969] Compiling libs/ardour/cycle_timer.cc
[ 844/1969] Compiling libs/ardour/thawlist.cc
[ 845/1969] Compiling libs/ardour/buffer.cc
In file included from include/boost/math/special_functions/fpclassify.hpp:15,
                 from ../libs/ardour/dsp_filter.cc:21:
include/boost/math/tools/config.hpp:28:6: warning: #warning "Boost.Math requires C++14" [-Wcpp]
In file included from include/boost/math/tools/promotion.hpp:28,
                 from include/boost/math/special_functions/math_fwd.hpp:28,
                 from include/boost/math/special_functions/fpclassify.hpp:20:
include/boost/math/tools/type_traits.hpp:208:12: error: ‘is_final’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:259:12: error: ‘remove_cv_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:261:12: error: ‘remove_const_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:263:12: error: ‘remove_volatile_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:265:12: error: ‘add_cv_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:267:12: error: ‘add_const_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:269:12: error: ‘add_volatile_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:273:12: error: ‘remove_reference_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:275:12: error: ‘add_lvalue_reference_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:277:12: error: ‘add_rvalue_reference_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:281:12: error: ‘remove_pointer_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:283:12: error: ‘add_pointer_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:287:12: error: ‘make_signed_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:289:12: error: ‘make_unsigned_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:293:12: error: ‘remove_extent_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:295:12: error: ‘remove_all_extents_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:299:12: error: ‘decay_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:301:12: error: ‘enable_if_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:303:12: error: ‘conditional_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:305:12: error: ‘common_type_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:307:12: error: ‘underlying_type_t’ has not been declared in ‘std’
include/boost/math/tools/type_traits.hpp:402:60: error: ‘is_final’ is not a member of ‘boost::math’; did you mean ‘is_final_v’?
include/boost/math/tools/type_traits.hpp:402:70: error: expected primary-expression before ‘>’ token
include/boost/math/tools/type_traits.hpp:402:73: error: ‘::value’ has not been declared
include/boost/math/tools/promotion.hpp:90:36: error: ‘common_type_t’ in namespace ‘boost::math’ does not name a template type; did you mean ‘common_type’?

In file included from ../libs/lua/LuaBridge/LuaBridge.h:71,
                 from ../libs/ardour/luabindings.cc:113:
In static member function ‘static R luabridge::FuncTraits<R (T::*)(P1), D>::call(T*, D, luabridge::TypeListValues<luabridge::TypeList<P2> >&) [with T = std::__cxx11::list<long int>; R = void; P1 = const long int&; D = void (std::__cxx11::list<long int>::*)(const long int&)]’,
    inlined from ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::__cxx11::list<long int>::*)(const long int&)]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:735:35:
../libs/lua/LuaBridge/detail/FuncTraits.h:261:22: warning: using a dangling pointer to an unnamed temporary [-Wdangling-pointer=]
  261 |     return (obj->*fp)(tvl.hd);
      |            ~~~~~~~~~~^~~~~~~~
In file included from ../libs/lua/LuaBridge/LuaBridge.h:70:
../libs/lua/LuaBridge/detail/TypeList.h: In static member function ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::__cxx11::list<long int>::*)(const long int&)]’:
../libs/lua/LuaBridge/detail/TypeList.h:127:74: note: unnamed temporary defined here
  127 |                                             ArgList <Tail, Start + 1> (L))
      |                                                                          ^
In static member function ‘static R luabridge::FuncTraits<R (T::*)(P1) const, D>::call(const T*, D, luabridge::TypeListValues<luabridge::TypeList<P2> >&) [with T = std::map<int, std::vector<_VampHost::Vamp::Plugin::Feature> >; R = long unsigned int; P1 = const int&; D = long unsigned int (std::map<int, std::vector<_VampHost::Vamp::Plugin::Feature> >::*)(const int&) const]’,
    inlined from ‘static int luabridge::CFunc::CallConstMember<MemFnPtr, ReturnType>::f(lua_State*) [with MemFnPtr = long unsigned int (std::map<int, std::vector<_VampHost::Vamp::Plugin::Feature> >::*)(const int&) const; ReturnType = long unsigned int]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:331:32:
../libs/lua/LuaBridge/detail/FuncTraits.h:458:22: warning: using a dangling pointer to an unnamed temporary [-Wdangling-pointer=]
  458 |     return (obj->*fp)(tvl.hd);
      |            ~~~~~~~~~~^~~~~~~~
../libs/lua/LuaBridge/detail/TypeList.h: In static member function ‘static int luabridge::CFunc::CallConstMember<MemFnPtr, ReturnType>::f(lua_State*) [with MemFnPtr = long unsigned int (std::map<int, std::vector<_VampHost::Vamp::Plugin::Feature> >::*)(const int&) const; ReturnType = long unsigned int]’:
../libs/lua/LuaBridge/detail/TypeList.h:127:74: note: unnamed temporary defined here
  127 |                                             ArgList <Tail, Start + 1> (L))
      |                                                                          ^
In static member function ‘static R luabridge::FuncTraits<R (T::*)(P1), D>::call(T*, D, luabridge::TypeListValues<luabridge::TypeList<P2> >&) [with T = std::vector<float>; R = void; P1 = const float&; D = void (std::vector<float>::*)(const float&)]’,
    inlined from ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::vector<float>::*)(const float&)]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:735:35:
../libs/lua/LuaBridge/detail/FuncTraits.h:261:22: warning: using a dangling pointer to an unnamed temporary [-Wdangling-pointer=]
  261 |     return (obj->*fp)(tvl.hd);
      |            ~~~~~~~~~~^~~~~~~~
../libs/lua/LuaBridge/detail/TypeList.h: In static member function ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::vector<float>::*)(const float&)]’:
../libs/lua/LuaBridge/detail/TypeList.h:127:74: note: unnamed temporary defined here
  127 |                                             ArgList <Tail, Start + 1> (L))
      |                                                                          ^
In static member function ‘static R luabridge::FuncTraits<R (T::*)(P1), D>::call(T*, D, luabridge::TypeListValues<luabridge::TypeList<P2> >&) [with T = std::vector<unsigned char>; R = void; P1 = const unsigned char&; D = void (std::vector<unsigned char>::*)(const unsigned char&)]’,
    inlined from ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::vector<unsigned char>::*)(const unsigned char&)]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:735:35:
../libs/lua/LuaBridge/detail/FuncTraits.h:261:22: warning: using a dangling pointer to an unnamed temporary [-Wdangling-pointer=]
  261 |     return (obj->*fp)(tvl.hd);
      |            ~~~~~~~~~~^~~~~~~~
../libs/lua/LuaBridge/detail/TypeList.h: In static member function ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::vector<unsigned char>::*)(const unsigned char&)]’:
../libs/lua/LuaBridge/detail/TypeList.h:127:74: note: unnamed temporary defined here
  127 |                                             ArgList <Tail, Start + 1> (L))
      |                                                                          ^
In static member function ‘static R luabridge::FuncTraits<R (T::*)(P1), D>::call(T*, D, luabridge::TypeListValues<luabridge::TypeList<P2> >&) [with T = std::vector<int>; R = void; P1 = const int&; D = void (std::vector<int>::*)(const int&)]’,
    inlined from ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::vector<int>::*)(const int&)]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:735:35:
../libs/lua/LuaBridge/detail/FuncTraits.h:261:22: warning: using a dangling pointer to an unnamed temporary [-Wdangling-pointer=]
  261 |     return (obj->*fp)(tvl.hd);
      |            ~~~~~~~~~~^~~~~~~~
../libs/lua/LuaBridge/detail/TypeList.h: In static member function ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::vector<int>::*)(const int&)]’:
../libs/lua/LuaBridge/detail/TypeList.h:127:74: note: unnamed temporary defined here
  127 |                                             ArgList <Tail, Start + 1> (L))
      |                                                                          ^
In static member function ‘static R luabridge::FuncTraits<R (T::*)(P1), D>::call(T*, D, luabridge::TypeListValues<luabridge::TypeList<P2> >&) [with T = std::vector<long int>; R = void; P1 = const long int&; D = void (std::vector<long int>::*)(const long int&)]’,
    inlined from ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::vector<long int>::*)(const long int&)]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:735:35:
../libs/lua/LuaBridge/detail/FuncTraits.h:261:22: warning: using a dangling pointer to an unnamed temporary [-Wdangling-pointer=]
  261 |     return (obj->*fp)(tvl.hd);
      |            ~~~~~~~~~~^~~~~~~~
../libs/lua/LuaBridge/detail/TypeList.h: In static member function ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::vector<long int>::*)(const long int&)]’:
../libs/lua/LuaBridge/detail/TypeList.h:127:74: note: unnamed temporary defined here
  127 |                                             ArgList <Tail, Start + 1> (L))
      |                                                                          ^
In static member function ‘static R luabridge::FuncTraits<R (T::*)(P1), D>::call(T*, D, luabridge::TypeListValues<luabridge::TypeList<P2> >&) [with T = std::vector<float*>; R = void; P1 = float* const&; D = void (std::vector<float*>::*)(float* const&)]’,
    inlined from ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::vector<float*>::*)(float* const&)]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:735:35:
../libs/lua/LuaBridge/detail/FuncTraits.h:261:22: warning: using a dangling pointer to an unnamed temporary [-Wdangling-pointer=]
  261 |     return (obj->*fp)(tvl.hd);
      |            ~~~~~~~~~~^~~~~~~~
../libs/lua/LuaBridge/detail/TypeList.h: In static member function ‘static int luabridge::CFunc::CallMember<MemFnPtr, void>::f(lua_State*) [with MemFnPtr = void (std::vector<float*>::*)(float* const&)]’:
../libs/lua/LuaBridge/detail/TypeList.h:127:74: note: unnamed temporary defined here
  127 |                                             ArgList <Tail, Start + 1> (L))
      |                                                                          ^
In file included from /nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/bits/shared_ptr_atomic.h:33,
                 from /nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/memory:81,
                 from /nix/store/wwnbp7vpdvjcizcw9n1qkcly0kr95dsl-glibmm-2.66.7-dev/include/glibmm-2.4/glibmm/variant.h:33,
                 from /nix/store/wwnbp7vpdvjcizcw9n1qkcly0kr95dsl-glibmm-2.66.7-dev/include/glibmm-2.4/glibmm/containerhandle_shared.h:23,
                 from /nix/store/wwnbp7vpdvjcizcw9n1qkcly0kr95dsl-glibmm-2.66.7-dev/include/glibmm-2.4/glibmm/arrayhandle.h:21,
                 from /nix/store/wwnbp7vpdvjcizcw9n1qkcly0kr95dsl-glibmm-2.66.7-dev/include/glibmm-2.4/glibmm.h:107,
                 from ../libs/ardour/luabindings.cc:23:
In member function ‘std::__atomic_base<_IntTp>::__int_type std::__atomic_base<_IntTp>::load(std::memory_order) const [with _ITp = long int]’,
    inlined from ‘int62_t::int62_t(const int62_t&)’ at ../libs/pbd/pbd/int62.h:64:44,
    inlined from ‘Temporal::timepos_t::timepos_t(const Temporal::timepos_t&)’ at ../libs/temporal/temporal/timeline.h:56:23,
    inlined from ‘Temporal::Range::Range(const Temporal::Range&)’ at ../libs/temporal/temporal/range.h:138:23,
    inlined from ‘ARDOUR::TimelineRange::TimelineRange(const ARDOUR::TimelineRange&)’ at ../libs/ardour/ardour/types.h:381:8,
    inlined from ‘static T luabridge::Stack<T>::get(lua_State*, int) [with T = ARDOUR::TimelineRange]’ at ../libs/lua/LuaBridge/detail/Userdata.h:726:51,
    inlined from ‘static int luabridge::CFunc::tableToListHelper(lua_State*, C*) [with T = ARDOUR::TimelineRange; C = std::__cxx11::list<ARDOUR::TimelineRange>]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:1242:15,
    inlined from ‘static int luabridge::CFunc::tableToListHelper(lua_State*, C*) [with T = ARDOUR::TimelineRange; C = std::__cxx11::list<ARDOUR::TimelineRange>]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:1234:14,
    inlined from ‘static int luabridge::CFunc::tableToList(lua_State*) [with T = ARDOUR::TimelineRange; C = std::__cxx11::list<ARDOUR::TimelineRange>]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:1256:36:
/nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/bits/atomic_base.h:501:31: warning: ‘long unsigned int __atomic_load_8(const volatile void*, int)’ writing 8 bytes into a region of size 0 overflows the destination [-Wstringop-overflow=]
  501 |         return __atomic_load_n(&_M_i, int(__m));
      |                ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
In static member function ‘static int luabridge::CFunc::tableToList(lua_State*) [with T = ARDOUR::TimelineRange; C = std::__cxx11::list<ARDOUR::TimelineRange>]’:
cc1plus: note: destination object is likely at address zero

Waf: Leaving directory `/build/ardour/build'
Build failed
 -> task in 'libardour' failed with exit status 1 (run with -v to display more information)
error: builder for '/nix/store/bhryai9rqql7d1lhcj94ms2znz7532b2-ardour-8.10.drv' failed with exit code 1;
       last 25 log lines:
       > In file included from /nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/bits/shared_ptr_atomic.h:33,
       >                  from /nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/memory:81,
       >                  from /nix/store/wwnbp7vpdvjcizcw9n1qkcly0kr95dsl-glibmm-2.66.7-dev/include/glibmm-2.4/glibmm/variant.h:33,
       >                  from /nix/store/wwnbp7vpdvjcizcw9n1qkcly0kr95dsl-glibmm-2.66.7-dev/include/glibmm-2.4/glibmm/containerhandle_shared.h:23,
       >                  from /nix/store/wwnbp7vpdvjcizcw9n1qkcly0kr95dsl-glibmm-2.66.7-dev/include/glibmm-2.4/glibmm/arrayhandle.h:21,
       >                  from /nix/store/wwnbp7vpdvjcizcw9n1qkcly0kr95dsl-glibmm-2.66.7-dev/include/glibmm-2.4/glibmm.h:107,
       >                  from ../libs/ardour/luabindings.cc:23:
       > In member function ‘std::__atomic_base<_IntTp>::__int_type std::__atomic_base<_IntTp>::load(std::memory_order) const [with _ITp = long int]’,
       >     inlined from ‘int62_t::int62_t(const int62_t&)’ at ../libs/pbd/pbd/int62.h:64:44,
       >     inlined from ‘Temporal::timepos_t::timepos_t(const Temporal::timepos_t&)’ at ../libs/temporal/temporal/timeline.h:56:23,
       >     inlined from ‘Temporal::Range::Range(const Temporal::Range&)’ at ../libs/temporal/temporal/range.h:138:23,
       >     inlined from ‘ARDOUR::TimelineRange::TimelineRange(const ARDOUR::TimelineRange&)’ at ../libs/ardour/ardour/types.h:381:8,
       >     inlined from ‘static T luabridge::Stack<T>::get(lua_State*, int) [with T = ARDOUR::TimelineRange]’ at ../libs/lua/LuaBridge/detail/Userdata.h:726:51,
       >     inlined from ‘static int luabridge::CFunc::tableToListHelper(lua_State*, C*) [with T = ARDOUR::TimelineRange; C = std::__cxx11::list<ARDOUR::TimelineRange>]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:1242:15,
       >     inlined from ‘static int luabridge::CFunc::tableToListHelper(lua_State*, C*) [with T = ARDOUR::TimelineRange; C = std::__cxx11::list<ARDOUR::TimelineRange>]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:1234:14,
       >     inlined from ‘static int luabridge::CFunc::tableToList(lua_State*) [with T = ARDOUR::TimelineRange; C = std::__cxx11::list<ARDOUR::TimelineRange>]’ at ../libs/lua/LuaBridge/detail/CFunctions.h:1256:36:
       > /nix/store/gnf3mv68i5g6jmabnbbncsar3kbg13zd-gcc-14-20241116/include/c++/14-20241116/bits/atomic_base.h:501:31: warning: ‘long unsigned int __atomic_load_8(const volatile void*, int)’ writing 8 bytes into a region of size 0 overflows the destination [-Wstringop-overflow=]
       >   501 |         return __atomic_load_n(&_M_i, int(__m));
       >       |                ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
       > In static member function ‘static int luabridge::CFunc::tableToList(lua_State*) [with T = ARDOUR::TimelineRange; C = std::__cxx11::list<ARDOUR::TimelineRange>]’:
       > cc1plus: note: destination object is likely at address zero
       > 
       > Waf: Leaving directory `/build/ardour/build'
       > Build failed
       >  -> task in 'libardour' failed with exit status 1 (run with -v to display more information)
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.9-xanmod1, NixOS, 25.05 (Warbler), 25.05beta742084.9e4d5190a948`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"home-manager, nixos, nur"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers

@magnetophon @mitchmindtree
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/audio/ardour/default.nix
+++ b/pkgs/applications/audio/ardour/default.nix
@@ -160,7 +160,7 @@ stdenv.mkDerivation rec {
   ] ++ lib.optionals videoSupport [ harvid xjadeo ];
 
   wafConfigureFlags = [
-    "--cxx11"
+    "--cxx17"
     "--docs"
     "--freedesktop"
     "--no-phone-home"

